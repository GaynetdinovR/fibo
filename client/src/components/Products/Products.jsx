import styles from "../../styles/components/Products.module.sass";
import H3 from "../../ui/H3.jsx";
import Product from "./components/Product.jsx";

const Products = ({ products }) => {
	const types = [
		{ type: "pizza", title: "Пиццы" },
		{ type: "pasta", title: "Пасты" },
		{ type: "salad", title: "Салаты" },
		{ type: "snack", title: "Закуски" },
		{ type: "drink", title: "Напитки" }
	];

	return (
		<section className={styles.products}>
			<div className={styles.products__categories}>
				{types.map((type, i) => (
					<div key={i} className={styles.products__category}>
						<H3 className={styles.products__title}>{type.title}</H3>
						<div className={styles.products__category_content}>
							{products
								?.filter((product) => product.type === type.type)
								.map((product, k) => (
									<Product key={k} product={product} />
								))}
						</div>
					</div>
				))}
			</div>
		</section>
	);
};

export default Products;
