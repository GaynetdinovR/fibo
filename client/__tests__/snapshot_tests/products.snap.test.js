import renderer from "react-test-renderer";
import { it, expect } from "@jest/globals";
import Products from "../../src/components/Products/Products.jsx";

const products = [
	{
		name: "test",
		desciption: "test",
		price: 299.99,
		is_new: true,
		img_url: "./content/pizzas/pizza_1.png",
		type: "pizza"
	},
	{
		name: "test",
		desciption: "test",
		price: 455.99,
		is_new: true,
		img_url: "./content/pastas/pasta_1.png",
		type: "pasta"
	},
];

it("renders correctly", () => {
	const tree = renderer.create(<Products products={products} />).toJSON();
	expect(tree).toMatchSnapshot();
});
