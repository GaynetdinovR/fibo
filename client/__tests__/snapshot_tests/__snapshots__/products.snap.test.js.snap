// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="products"
>
  <div
    className="products__categories"
  >
    <div
      className="products__category"
    >
      <h4
        className="products__title h3"
      >
        Пиццы
      </h4>
      <div
        className="products__category_content"
      >
        <div
          className="products__product"
        >
          <div
            className="product__top_side"
          >
            <div
              className="product__img"
            >
              <img
                alt="product_img"
                src="./content/pizzas/pizza_1.png"
              />
              <div
                className="product__new_tag"
              >
                NEW
              </div>
            </div>
            <div
              className="product__info"
            >
              <h4
                className="product__name h4"
              >
                test
              </h4>
              <p
                className="product__text text"
              >
                 
                 
              </p>
            </div>
          </div>
          <div
            className="product__bottom_side"
          >
            <span
              className="product__price"
            >
              299.99
               ₽
            </span>
            <button
              className="product__to_cart_btn default_button"
            >
              В корзину
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="products__category"
    >
      <h4
        className="products__title h3"
      >
        Пасты
      </h4>
      <div
        className="products__category_content"
      >
        <div
          className="products__product"
        >
          <div
            className="product__top_side"
          >
            <div
              className="product__img"
            >
              <img
                alt="product_img"
                src="./content/pastas/pasta_1.png"
              />
              <div
                className="product__new_tag"
              >
                NEW
              </div>
            </div>
            <div
              className="product__info"
            >
              <h4
                className="product__name h4"
              >
                test
              </h4>
              <p
                className="product__text text"
              >
                 
                 
              </p>
            </div>
          </div>
          <div
            className="product__bottom_side"
          >
            <span
              className="product__price"
            >
              455.99
               ₽
            </span>
            <button
              className="product__to_cart_btn default_button"
            >
              В корзину
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="products__category"
    >
      <h4
        className="products__title h3"
      >
        Салаты
      </h4>
      <div
        className="products__category_content"
      />
    </div>
    <div
      className="products__category"
    >
      <h4
        className="products__title h3"
      >
        Закуски
      </h4>
      <div
        className="products__category_content"
      />
    </div>
    <div
      className="products__category"
    >
      <h4
        className="products__title h3"
      >
        Напитки
      </h4>
      <div
        className="products__category_content"
      />
    </div>
  </div>
</section>
`;
